src/apis/logs_api.ts

```diff
@@ -1,25 +1,116 @@
 
 import { Web3Utils } from "../../utils/web3_utils";
 
+export interface GetEventOptions {
+    fromBlock?: number;
+    toBlock?: number;
+    limit?: number;
+}
+
 export class LogsAPI {
+    private readonly web3: Web3;
     private web3Utils: Web3Utils;
     private contracts: ContractsAPI;
 
     constructor(web3: Web3, contracts: ContractsAPI) {
         this.web3Utils = new Web3Utils(web3);
+        this.web3 = web3;
         this.contracts = contracts;
+    }
 
-        // We need to configure the ABI Decoder in order to pull out relevant logs.
-        ABIDecoder.addABI(DebtKernel.abi);
-        ABIDecoder.addABI(RepaymentRouter.abi);
+    /**
+     * @example
+     * // Get all "LogLoanOrderFilled" events between blocks 0 and 300.
+     * await dharma.logs.get(["LogLoanOrderFilled"], { from: 0, to: 300 });
+     *
+     * // Get the last 10 "LogLoanOrderFilled" events between blocks 0 and 300.
+     * await dharma.logs.get(["LogLoanOrderFilled"], { from: 0, to: 300, limit: 10 });
+     *
+     * // Get the last 10 "LogLoanOrderFilled" events.
+     * await dharma.logs.get(["LogLoanOrderFilled"], { limit: 10 });
+     *
+     * // Get the last 10 "LogLoanOrderFilled" or "LogLoanOrderCancelled" events that occurred.
+     * await dharma.logs.get(["LogLoanOrderFilled", "LogLoanOrderCancelled"], { limit: 10 });
+     *
+     * // Get all of the "LogLoanOrderFilled" events.
+     * await dharma.logs.get(["LogLoanOrderFilled"]);
+     *
+     * // Get all of the "LogLoanOrderFilled" events.
+     * await dharma.logs.get("LogLoanOrderFilled");
+     *
+     * @param {string | string[]} eventNames
+     * @param {GetEventOptions} options
+     * @returns {Promise<any>}
+     */
+    public async get(eventNames: string | string[], options: GetEventOptions = {}): Promise<any> {
+        const { fromBlock, limit, toBlock } = options;
+
+        eventNames = _.castArray(eventNames);
+
+        const eventToContract = await this.getEventToContractsMap();
+
+        let events = [];
+
+        await Promise.all(
+            eventNames.map(async (eventName) => {
+                const contractWrapper = eventToContract[eventName];
+                const contract = this.web3.eth
+                    .contract(contractWrapper.abi)
+                    .at(contractWrapper.address);
+
+                return new Promise((resolve) => {
+                    const eventsWatcher = contract.allEvents({
+                        fromBlock: fromBlock || 0,
+                        toBlock: toBlock || "latest",
+                    });
+
+                    eventsWatcher.get((error, logs) => {
+                        const filteredEvents = _.filter(logs, (log) => log.event === eventName);
+
+                        events = events.concat(filteredEvents);
+
+                        resolve();
+                    });
+
+                    eventsWatcher.stopWatching();
+                });
+            }),
+        );
+
+        if (_.isNumber(limit)) {
```

Comment:
Doesn't feel necessary to test for number here. If the user specified 0 as the limit, we shouldn't be doing any work. Just return.

Reply:
The possible cases are: 0, non-zero number, and null. If not null: don't limit, if 0: limit, if greater than 0: limit. So not checking number would make three branches:

if (limit === 0)
  return [];
if (!limit) 
  return everything
else
  return data with limit

Reply:
```
if (limit) {
  return data with limit
} else {
  return everything
}
```